### PR TITLE
Support V2 (triggered and orchestratedEmail) events

### PR DESCRIPTION
**Pull request by fleetwoods**

### CODE REVIEW

File: test/aws/dynamo/DynamoSpec.scala [hunk 1]
```diff
@@ -52,6 +71,35 @@ class DynamoSpec extends FlatSpec
       triggeredResult.traceToken shouldBe triggeredEvent.metadata.traceToken
       triggeredResult.timestamp shouldBe OffsetDateTime.parse(triggeredEvent.metadata.createdAt)
       triggeredResult.events.length shouldBe 1
+      triggeredResult.events.head.asInstanceOf[TriggeredEvent].event shouldBe triggeredEvent
```
Comment (cb372) [inline test/aws/dynamo/DynamoSpec.scala]: I think you can write this as `triggeredResult.events.head should be a [TriggeredEvent]`

File: test/util/TestGenerators.scala [hunk 2]
```diff
@@ -2,8 +2,10 @@ package util
 
 import java.util.UUID
 
+import com.ovoenergy.comms.model.TemplateData
 import org.scalacheck.{Arbitrary, Gen}
 import org.scalacheck.Shapeless._
+import org.scalacheck.derive.Recursive
```
Comment (cb372) [inline test/util/TestGenerators.scala]: some stray imports here

File: test/aws/dynamo/DynamoSpec.scala (hunk 1 above)
Comment (fleetwoods) [inline test/aws/dynamo/DynamoSpec.scala]: I wanted it to test that is actually encoded/decoded (round trip) from the event and back, i.e. specifically in the case of the V2 event the template data

**Comment by ovo-comms-circleci** [comment]
This PR was successfully deployed to production by [CircleCI build 54](https://circleci.com/gh/ovotech/comms-audit-log/54) at 2017-01-31 17:47 GMT.

**Pull request merged**